### PR TITLE
31 - Added bot admin commands

### LINKED ISSUE #31: Bot admins
The bot should keep track of a list of admins that determines which non-discord-admin users are allowed to execute admin bot commands

### PR DESCRIPTION
**Pull request by woutermeuwis**
closes #31

### CODE REVIEW

File: commands/admin.js [hunk 1]
```diff
@@ -0,0 +1,154 @@
...
+
+  // Build reply
+  const memberManager = interaction.guild.members
+  const admins = await memberManager.fetch({ user: guild.admins })
+  let response = 'De bot admins voor deze server zijn:'
+  for (const [, admin] of admins) { response += `\n${admin.user.username}` }
+  await interaction.reply(response)
+}
+
+async function addAdmin (interaction) {
```
Comment (SelleslaghGianni) [inline commands/admin.js:38]: Produces this error when adding someone as admin without having the permission to do so and then crashes out. The bot replies with the "da gade gy ni bepalen"-gif, but still adds that id to the admins list.
Error:
```
{
  level: 'info',
  message: 'JSON data saved to file',
  timestamp: '2022-04-18T15:38:56.095Z'
}
{
  level: 'error',
  message: 'DiscordAPIError: Interaction has already been acknowledged.',
  timestamp: '2022-04-18T15:38:56.477Z',
  stack: 'Error: DiscordAPIError: Interaction has already been acknowledged.\n' +
    '    at Object.error (/home/mintage/dev/nerdlandbot_js/helpers/logger.js:43:7)\n' +
    '    at Client.<anonymous> (/home/mintage/dev/nerdlandbot_js/nerdlandbot.js:98:9)\n' +
    '    at processTicksAndRejections (node:internal/process/task_queues:96:5)'
}
```

**Review by SelleslaghGianni — CHANGES_REQUESTED** [review]

File: commands/admin.js (hunk 1 above)
Comment (woutermeuwis) [reply commands/admin.js:38]: good catch

File: commands/admin.js [hunk 2]
```diff
@@ -4,15 +4,19 @@ const { getGuild, saveGuild } = require('../helpers/guildData')
...
+async function verifyAdminAsync (interaction) {
   const user = interaction.member
   const guild = await getGuild(interaction.guildId)
 
   const isGuildAdmin = user.permissions.has(Permissions.FLAGS.ADMINISTRATOR)
   const isBotAdmin = guild.admins.includes(user.id)
 
   if (!isGuildAdmin && !isBotAdmin) {
-    await interaction.reply('https://gph.is/g/4w8PDNj')
+    if (interaction.isdeferred || interaction.replied) {
```
Comment (SelleslaghGianni) [inline commands/admin.js:15]: There is an issue with this line of code. 
This is the error it causes:
```
error:   2022-04-29T14:35:56.225Z | Error: Error [INTERACTION_ALREADY_REPLIED]: The reply to this interaction has already been sent or deferred.
    at Object.error (/home/mintage/dev/nerdlandbot_js/helpers/logger.js:43:7)
    at Client.<anonymous> (/home/mintage/dev/nerdlandbot_js/nerdlandbot.js:98:9)
    at processTicksAndRejections (node:internal/process/task_queues:96:5)
/home/mintage/dev/nerdlandbot_js/node_modules/discord.js/src/structures/interfaces/InteractionResponses.js:90
    if (this.deferred || this.replied) throw new Error('INTERACTION_ALREADY_REPLIED');
                                             ^

Error [INTERACTION_ALREADY_REPLIED]: The reply to this interaction has already been sent or deferred.
    at CommandInteraction.reply (/home/mintage/dev/nerdlandbot_js/node_modules/discord.js/src/structures/interfaces/InteractionResponses.js:90:46)
    at Client.<anonymous> (/home/mintage/dev/nerdlandbot_js/nerdlandbot.js:100:23)
    at processTicksAndRejections (node:internal/process/task_queues:96:5) {
  [Symbol(code)]: 'INTERACTION_ALREADY_REPLIED'
}
```
The fix is simple:
```if (interaction.deferred || interaction.replied) {```

**Review by SelleslaghGianni — CHANGES_REQUESTED** [review]
If the changes get made you've got an approval :)

**Review by SelleslaghGianni — APPROVED** [review]
LGTM :)

**Pull request merged**